README.md

```diff
@@ -1,11 +1,250 @@
+
+:warning: **Important** :warning:
+
+When running the `convert_dataset_hf.py` or `convert_dataset_json.py` in the steps above, you will have to specify that you are using the Replit tokenizer by passing in the argument ` --tokenizer replit/replit-code-v1-3b`.
+A key thing needed here, due to a hardcoded logic in `composer` is that you will have to edit the `convert_dataset_hf.py` by passing the `trust_remote_code=True` kwarg to the `AutoTokenizer.from_pretrained` call when the tokenizer is loaded on Line 315.
+
+**Testing Your Converted Dataset**
+
+To test the converted dataset and check that its working with the dataloader, you can follow the [Test the Dataloader](https://github.com/mosaicml/llm-foundry/tree/main/scripts/train#test-the-dataloader) section in LLM Foundry docs. 
+
+#### (2) Define a Run Configuration YAML with the Replit Models
+
+To train with LLM Foundry, you need to define a run configuration yaml. This yaml defines the model, training dataset, eval dataset and metric, training parameters and more.
+
+**Using the Replit Models**
+
+For any config YAML you define to train/tune with LLM Foundry, you can plug in and use the Replit model by replacing  the model and tokenizer keys in your YAML as follows:
+```
+...
+model:
+  name: hf_causal_lm
+  pretrained: true
+  pretrained_model_name_or_path: replit/replit-code-v1-3b
+  config_overrides:
+    attn_config:
+      attn_impl: triton
+      attn_uses_sequence_id: false
+
+tokenizer:
+  name: replit/replit-code-v1-3b
+  kwargs:
+    model_max_length: ${max_seq_len}
+    trust_remote_code: true
+...
+```
+
+This will load our model with its weights from Huggingface for your config.
+
+#### (3) Running Training with LLM Foundry and Composer
+
+
+After having converted your dataset and defined a run configuration yaml, you can run training with LLM Foundry.
+
+Follow the [How to Start Training](https://github.com/mosaicml/llm-foundry/tree/main/scripts/train#how-to-start-training) section in the LLM Foundry docs to run training. The section shows you how to run single-node and multi-node training.
+Effectively, you will run the `scripts/train/train.py` training script in the LLM Foundry repo with the defined configuration yaml using a command like `composer train/train.py <configuration_yaml_path> <extra_args>`.
+
+:warning: **Important** :warning:
+
+There is some hardcoded logic in Composer that we need to circumvent right now to ensure we can save out checkpoints. 
+In the `scripts/train/train.py` training script, add the line `model.tokenizer = None` just after the model is initialized and the before the train dataloader is set up, i.e. Line 147. 
+This is effectively ensures that we don't save out the tokenizer with the checkpoint state. This is needed because composer currently cannot handle saving checkpoints with tokenizers that have *.py files. 
```

Comment:
typo: "This is effectively ensures"

Reply:
```suggestion
This effectively ensures that we don't save out the tokenizer with the checkpoint state. This is needed because composer currently cannot handle saving checkpoints with tokenizers that have *.py files. 
```